adapter-base/src/main/java/org/eclipse/hono/adapter/auth/device/jwt/CredentialsParser.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.adapter.auth.device.jwt;
+
+import java.net.HttpURLConnection;
+import java.util.Optional;
+
+import org.eclipse.hono.client.ClientErrorException;
+import org.eclipse.hono.util.CredentialsConstants;
+
+import io.jsonwebtoken.Claims;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * A service for extracting and parsing tenant and device information.
+ *
+ */
+public interface CredentialsParser {
+
+    /**
+     * Extracts the tenantId and authId from JWS Claims.
+     *
+     * @param claims A JsonObject containing the JWS Claims.
+     * @return A JsonObject containing the tenantId and authId extracted from the JWS Claims.
```

Comment:
IMHO you should be more specific here regarding the property names to expect in the returned JSON ...

Reply:
yeah makes sense, I have adjusted the documentation to better reflect the expected property names.